test/integration/integration_test.go

```diff
@@ -432,15 +438,35 @@ func (i *integrationTest) exerciseInjectionBuild(tag, imageName string, injectio
 	i.fileExists(containerID, "/sti-fake/relative-secret-delivered")
 
 	// Make sure the injected file does not exists in resulting image
-	files, err := util.ExpandInjectedFiles(fs.NewFileSystem(), injectionList)
+	testFs := fs.NewFileSystem()
+	files, err := util.ListFilesToTruncate(testFs, injectionList)
 	if err != nil {
 		t.Errorf("Unexpected error: %v", err)
 	}
 	for _, f := range files {
-		if exitCode := i.runInImage(tag, "test -s "+f); exitCode == 0 {
+		if err = i.testFile(tag, f); err == nil {
 			t.Errorf("The file %q must be empty", f)
 		}
 	}
+	if keptVolume != nil {
+		keptFiles, err := util.ListFiles(testFs, *keptVolume)
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+		for _, f := range keptFiles {
+			if err = i.testFile(tag, f); err != nil {
+				t.Errorf("The file %q must not be empty", f)
+			}
+		}
+	}
+}
+
+func (i *integrationTest) testFile(tag, path string) error {
+	exitCode := i.runInImage(tag, "test -s "+path)
+	if exitCode != 0 {
+		return fmt.Errorf("file %s does not exist in the container %s", path, tag)
```

Comment:
not strictly true, since `test -s` also returns "1" if the file exists w/ size zero.  should probably fix this message so it doesn't confuse someone.